src/bin/radclient.c

```diff
@@ -1027,13 +1028,12 @@ static int recv_one_packet(int wait_time)
 	max_fd = fr_packet_list_fd_set(packet_list, &set);
 	if (max_fd < 0) exit(1); /* no sockets to listen on! */
 
-	tv.tv_sec = (wait_time <= 0) ? 0 : wait_time;
-	tv.tv_usec = 0;
+	wait_time = (wtime <= 0) ? 0 : fr_time_delta_from_sec(wtime);
```

Comment:
The type of wtime is a type that holds the time in nanoseconds, but you're converting it to seconds here, but wait_time is also a time delta in nanoseconds, and you convert the delta to a timeval later... so the conversion from delta to seconds isn't needed, and will result in huge waiting periods.